Fix/dependencies for php 8.1.x

lukaslangen opened the pull request:
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | yes
| BC Break      | unsure
| New Feature   | no
| RFC           | yes
| QA            | no

### Description

See #59

Commit:
Fix installer for usage with php 8.1

Removed packages:
    - laminas/laminas-auradi-config
    - mezzio/mezzio-aurarouter
    - northwoods/container

Updated packages:
    - jsoumelidis/zend-sf-di-config from ^0.4 to ^0.5
    - mezzio/mezzio-tooling from ^1.4.0 to ^2.0

Also removed all usages and mentions of auradi aurarouter and auryn

Signed-off-by: Lukas Langen <lukas@langen.dev>

Commit:
Use laminas cli instead of mezzio

Signed-off-by: Lukas Langen <lukas@langen.dev>

Ocramius commented on src/App/src/Handler/HomePageHandlerFactory.php:
```diff
@@ -20,6 +20,8 @@ public function __invoke(ContainerInterface $container): RequestHandlerInterface
             ? $container->get(TemplateRendererInterface::class)
             : null;
 
+        // @codingStandardsIgnoreStart
```
What was the reported issue here? :thinking:

lukaslangen replied:
```FILE: src/App/src/Handler/HomePageHandlerFactory.php
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------
FOUND 1 ERROR AFFECTING 1 LINE
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------
 23 | ERROR | [x] Class name referenced via call of function get_class().
    |       |     (SlevomatCodingStandard.Classes.ModernClassNameReference.ClassNameReferencedViaFunctionCall)
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------
PHPCBF CAN FIX THE 1 MARKED SNIFF VIOLATIONS AUTOMATICALLY
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------

Time: 81ms; Memory: 10MB
```

arueckauer replied:
`@codingStandardsIgnoreStart` is deprecated, `phpcs:disable` should be used. For a single line `phpcs:ignore` can be used, which causes a little less noise.
https://github.com/squizlabs/PHP_CodeSniffer/wiki/Advanced-Usage#ignoring-parts-of-a-file

Ocramius reviewed (approved):
LGTM.

@weierophinney @boesing can you check? I think having less integration work here is a good idea.

brandonsavage commented:
I think removing Aura is a good call, since that project hasn't been updated in some time and doesn't appear that it will be. I'm migrating clients to Mezzio away from Aura, because it's not actively supported.

Ocramius commented:
CS fixes are still running on 7.3: perhaps we just need to drop support for that?

weierophinney reviewed (changes requested):
I agree with @Ocramius — bump the minimum supported PHP version to 7.4. From there, also do the following:

- Bump laminas-coding-standard to `^2.3` (and fix any issues raised).
- Bump PHPUnit to `^9.5.5`.
- Check other dependencies to determine which version first offers PHP 8.1 support, and bump to that when possible.
- Audit the cs-check and cs-fix scripts; is the `DealerDirect....Plugin::run` line really required for these anymore?

Before we release, I'd also like to see a separate PR that replaces phpstan with psalm, so that this package is consistent with our other packages.

Commit:
Bump coding standard version constraint

Signed-off-by: Lukas Langen <lukas@langen.dev>

lukaslangen committed:
Bump phpunit version constraint

Signed-off-by: Lukas Langen <lukas@langen.dev>

lukaslangen committed:
Bump minimum php version to ^7.4

Signed-off-by: Lukas Langen <lukas@langen.dev>

lukaslangen committed:
Bump laminas-component-installer version constraint

Signed-off-by: Lukas Langen <lukas@langen.dev>

lukaslangen committed:
Bump laminas-config-aggregator version constraint

Signed-off-by: Lukas Langen <lukas@langen.dev>

lukaslangen committed:
Bump laminas-diactoros version constraint

Signed-off-by: Lukas Langen <lukas@langen.dev>

lukaslangen committed:
Bump laminas-stdlib version constraint

Signed-off-by: Lukas Langen <lukas@langen.dev>

lukaslangen committed:
Remove laminas-zendframework-bridge

Signed-off-by: Lukas Langen <lukas@langen.dev>

lukaslangen committed:
Bump mezzio version constraint

Signed-off-by: Lukas Langen <lukas@langen.dev>

lukaslangen committed:
Bump mezzio-helpers version constraint

Signed-off-by: Lukas Langen <lukas@langen.dev>

lukaslangen commented:
Okay. I've added the specifically requested new version constraints and checked all version constraints for `require` and adjusted them accordingly.

Also on the way I removed `laminas/laminas-zendframework-bridge` as per #51 

As for the cs-fix stuff: I don't have the time right now to go more deeply into this, but it seems, that this line is necessary to install *all* coding standards found in the vendor directory. It seems especially necessary for CI (at least after my understanding of the package's documentation, see [here](https://github.com/Dealerdirect/phpcodesniffer-composer-installer#calling-the-plugin-directly)

If it still should be removed, please just tell me and I'll do it as soon, as I have time for it.

weierophinney commented:
> As for the cs-fix stuff: I don't have the time right now to go more deeply into this, but it seems, that this line is necessary to install _all_ coding standards found in the vendor directory. It seems especially necessary for CI (at least after my understanding of the package's documentation, see [here](https://github.com/Dealerdirect/phpcodesniffer-composer-installer#calling-the-plugin-directly)

What's odd is we literally don't have that in any other package we ship, and CI works just fine. :)

I have time today, so I'll see if I can push that change and fixes for the failing unit tests on 7.4 with lowest deps.

weierophinney committed:
qa: fix issues with composer install, coding standards

As of Composer 2.2, we need to add an "allow-plugins" configuration directive with a list of plugins we allow (see https://getcomposer.org/doc/06-config.md#allow-plugins).
Additionally, the installation was not originally performed with PHP 7.4, which meant that when running `--prefer-lowest`, we would run into errors.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
fix: remove unneeded (?) lines from cs-check/cs-fix scripts

These are not needed in any of our packages, and I've not needed them in any sites with which I use laminas-coding-standard or phpcs.
If CI fails, I'll revert.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>
